src/ast_selectors.cpp

```diff
@@ -154,19 +155,17 @@ namespace Sass {
       return false;
     }
 
-    // would like to replace this without stringification
+    // replaced compare without stringification
     // https://github.com/sass/sass/issues/2229
-    // SimpleSelectorSet lset, rset;
-    std::set<std::string> lset, rset;
+    SelectorSet lset, rset;
 
     if (lbase && rbase)
     {
-      if (lbase->to_string() == rbase->to_string()) {
-        for (size_t i = 1, L = length(); i < L; ++i)
-        { lset.insert((*this)[i]->to_string()); }
-        for (size_t i = 1, L = rhs->length(); i < L; ++i)
-        { rset.insert((*rhs)[i]->to_string()); }
-        return includes(rset.begin(), rset.end(), lset.begin(), lset.end());
+      if (*lbase == *rbase) {
+        // create ordered sets for includes query
+        lset.insert(this->begin(), this->end());
```

Comment:
`std::set` can be constructed directly from a range:

```c++
SelectorSet lset(lset.begin(), lset.end());
SelectorSet rset(rset.begin(), rset.end());
```

Reply:
Thx for the top, I'll see that I can incorporate it in a later commit!

Reply:
Need to check if it is possible though, not sure we "always" want to construct it with those items.